### ISSUE TITLE
Flaky unit test in spire/pkg/agent/endpoints/sdsv3: TestStreamSecrets

### ISSUE THREAD

**Issue by amartinezfayo** [spiffe/spire#3229]
```
--- FAIL: TestStreamSecrets (0.07s)
    --- FAIL: TestStreamSecrets/Unknown_resource (0.00s)
        handler_test.go:632: 
            	Error Trace:	/home/runner/work/spire/spire/pkg/agent/endpoints/sdsv3/assertions.go:50
            	            				/home/runner/work/spire/spire/pkg/agent/endpoints/sdsv3/handler_test.go:632
            	Error:      	Status code="Internal" msg="malformed header: missing HTTP status; malformed header: missing HTTP content-type" does not match code="InvalidArgument" msg="workload is not authorized for the requested identities [\"spiffe://domain.test/WHATEVER\"]"
            	Test:       	TestStreamSecrets/Unknown_resource
FAIL
FAIL	github.com/spiffe/spire/pkg/agent/endpoints/sdsv3	0.140s
```

**Comment by howardjohn** [comment]
Hello from Istio... ran into this googling an error message in our test failure -- also in an SDS test, although it seems more likely a grpc issue (library, or usage of) than something sds-specific. I'll update if we root cause

**Comment by MarcosDY** [comment]
[grpc/grpc-go/issues/5511](https://github.com/grpc/grpc-go/issues/5511) was released on [grpc/grpc-go/pull/5513](https://github.com/grpc/grpc-go/pull/5513) and released on grpc v1.49.0.
